zerver/views/streams.py

```diff
@@ -175,13 +188,15 @@ def update_subscriptions_backend(request, user_profile,
 
 @authenticated_json_post_view
 def json_remove_subscriptions(request, user_profile):
+    # type: (HttpRequest, UserProfile) -> HttpResponse
     return remove_subscriptions_backend(request, user_profile)
 
 @has_request_variables
 def remove_subscriptions_backend(request, user_profile,
                                  streams_raw = REQ("subscriptions", validator=check_list(check_string)),
                                  principals = REQ(validator=check_list(check_string), default=None)):
 
+    # type: (HttpRequest, UserProfile, Any, Any) -> HttpResponse
```

Comment:
Here, if "Any"' is replaced by REQ, following are the error messages:
`
zerver/views/streams.py: note: In function "remove_subscriptions_backend":
zerver/views/streams.py:201: error: Argument 1 to "set" has incompatible type "REQ"; expected Iterable[None] 

zerver/views/streams.py:207: error: Argument 1 to "list_to_streams" has incompatible type "REQ"; expected List[str]

zerver/views/streams.py:218: error: Iterable expected

zerver/views/streams.py:218: error: "REQ" has no attribute "__iter__"

`
To understand the reason of above errors, I explored 'REQ' class and 'has_request_variables'.

Function will be called as:
` remove_subscriptions_back-end(request,UserProfile)`
This function operates on the request objects which have the same request variables which are provided in "**kwargs" as `REQ` objects during implementation of this function. Therefore when this function is called, only two arguments would be provided by default - request and UserProfile.


1). What does this decorator do?
     It extracts the specified request variables  from the request object when the function is called. These variables must be present in the request object by default. So when After the function is decorated, it will have different parameters according to the request variables(they might be list of strings and int...thus "**kwargs changed after decoration") and there are no more REQ objects which will be passed to the function.
     
     Therefore at run-time no errors will be raised when this function is executed.
     

When mypy runs on this file, the function is not decorated and so the argument to functions like "set" and "list_to_streams" are not compatible. The arguments would become compatible only after they have been decorated,which occurs when function is executed at runtime.